src/ApiService/ApiService/Program.cs

```diff
@@ -130,19 +145,33 @@ public class LoggingMiddleware : IFunctionsWorkerMiddleware {
                 .AddSingleton<EntityConverter>()
                 .AddSingleton<IServiceConfig>(configuration)
                 .AddSingleton<IStorage, Storage>()
-                .AddSingleton<ILogSinks, LogSinks>()
                 .AddHttpClient()
                 .AddMemoryCache()
                 .AddAzureAppConfiguration();
 
                 _ = services.AddFeatureManagement();
             })
+            .ConfigureLogging(loggingBuilder => {
+                loggingBuilder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider, OneFuzzLoggerProvider>(
+                    x => {
+                        var appInsightsConnectionString = $"InstrumentationKey={configuration.ApplicationInsightsInstrumentationKey}";
+                        var tc = new ApplicationInsights.TelemetryClient(new ApplicationInsights.Extensibility.TelemetryConfiguration() { ConnectionString = appInsightsConnectionString });
+                        return new OneFuzzLoggerProvider(new List<TelemetryConfig>() { new TelemetryConfig(tc) });
+                    }
+                    ));
+            })
             .ConfigureFunctionsWorkerDefaults(builder => {
-                builder.UseAzureAppConfiguration();
                 builder.UseMiddleware<LoggingMiddleware>();
+                builder.UseAzureAppConfiguration();
+
+                //this is a must, to tell the host that worker logging is done by us
+                builder.Services.Configure<WorkerOptions>(workerOptions => workerOptions.Capabilities["WorkerApplicationInsightsLoggingEnabled"] = bool.TrueString);
                 builder.AddApplicationInsights(options => {
-                    options.ConnectionString = $"InstrumentationKey={configuration.ApplicationInsightsInstrumentationKey}";
+                    options.DeveloperMode = false;
+                    options.EnableDependencyTrackingTelemetryModule = false;
```

Comment:
Disabling dependency tracking?

Reply:
we can enable - but it's sooooo noisy 

I will enable, but it will be a lot

Reply:
We already have it enabled in the production deployment (which is how we noticed #3156, for example)